runtime/src/cost_calculation_metrics.rs

```diff
@@ -0,0 +1,103 @@
+use {
+    solana_program_runtime::timings::ProgramTiming,
+    solana_sdk::pubkey::Pubkey,
+    std::time::{Duration, Instant},
+};
+
+/// To prevent too many writes to influx too often, MetricsMeter meters
+/// up to `max_number_of_writes` of writes within `time_window_duration`.
+/// New time window starts from the first write attempt outside of previous
+/// time window. See tests for examples.
+#[derive(Debug)]
+struct MetricsMeter {
+    max_number_of_writes: usize,
+    time_window_duration: Duration,
+    accumulated_number_of_writes: usize,
+    time_window_start: Instant,
+}
+
+impl MetricsMeter {
+    pub fn new(max_number_of_writes: usize, time_window_duration: Duration) -> Self {
+        Self {
+            max_number_of_writes,
+            time_window_duration,
+            accumulated_number_of_writes: 0,
+            time_window_start: Instant::now(),
+        }
+    }
+
+    pub fn take(&mut self) -> bool {
+        if self.time_window_start.elapsed() > self.time_window_duration {
+            // To start a new window, reset counters
+            self.accumulated_number_of_writes = 0;
+            self.time_window_start = Instant::now();
+        }
+        self.accumulated_number_of_writes = self.accumulated_number_of_writes.saturating_add(1);
+        self.accumulated_number_of_writes <= self.max_number_of_writes
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct CostCalculationMetrics {
+    metrics_meter: MetricsMeter,
+}
+
+impl CostCalculationMetrics {
+    pub fn new(max_number_of_writes: usize, time_window_duration: Duration) -> Self {
+        Self {
+            metrics_meter: MetricsMeter::new(max_number_of_writes, time_window_duration),
+        }
+    }
+
+    pub fn report(
+        &mut self,
+        program_id: &Pubkey,
+        program_timing: &ProgramTiming,
+        calculated_units: u64,
+    ) {
+        if self.metrics_meter.take() {
```

Comment:
use `MatricsMeter` to limit number of metrics publishing is up to 100 per sec (100 is defined below)